### PR TITLE
feat: add charging for all memory copy/init operations

### LINKED ISSUE #1075: Charge for table growing
We don't curently charge for table growing, but we should. According to the wasmtime docs, each table element requires one usize of memory (host memory).

Proposal: Charge 8 bytes per table element.

1. Instrument and charge for zeroing in wasm.
2. Account for the expanded table memory in wasm.

### PR DESCRIPTION
**Pull request by Stebalien**
I'm not actually charging anything right now, but this lets us charge for:

1. Memory copying.
2. Memory "filling" (where I'm treating all memory expansion as "filling").

fixes #1075

### CODE REVIEW

**Comment by Stebalien** [comment]
Based on #1243.

File: fvm/src/gas/price_list.rs [hunk 1]
```diff
@@ -1047,22 +1076,30 @@ impl Rules for WasmGasPrices {
...
-                match expansion_cost
-                    .try_into().ok() // zero or not zero.
-                {
-                    Some(cost) => InstructionCost::Linear(0, cost),
-                    None => InstructionCost::Fixed(0),
-                }
-            }),
+            // Memory related instructions...
+            Operator::TableInit { .. } | Operator::TableCopy { .. } => linear_cost(
+                Gas::zero(),
```
Comment (magik6k) [inline fvm/src/gas/price_list.rs:1081]: No base?

**Review by magik6k — APPROVED** [review]

File: fvm/src/gas/price_list.rs (hunk 1 above)
Comment (Stebalien) [reply fvm/src/gas/price_list.rs:1081]: Hm. Yeah, I'll make it configurable. These are all zero right now anyways.

Comment (Stebalien) [reply fvm/src/gas/price_list.rs]: Done.

**Comment by Stebalien** [comment]
This further reduces gas fees by 504. But it shouldn't change anything.

**Comment by Stebalien** [comment]
Oh.... It's because we were charging a flat cost before. Let me fix that.

**Comment by Stebalien** [comment]
Ok, now this costs 4 more than master... 8 more than the previous PR.

That's _probably_ because we're charging for tables.

**Comment by Stebalien** [comment]
Yeah, we're now charging for exactly one more instruction than we did before.

**Comment by Stebalien** [comment]
Ah... fixed.

**Pull request merged**